src/Accounts/Accounts/Common/AzureContextModificationCmdlet.cs

```diff
@@ -109,17 +111,30 @@ protected virtual ContextModificationScope GetContextModificationScope()
         /// <summary>
         /// Initialize the profile provider based on the autosave setting
         /// </summary>
-        internal void InitializeProfileProvider(bool useAutoSaveProfile = false)
+        internal bool InitializeProfileProvider(bool useAutoSaveProfile = false)
         {
 #if DEBUG
             if (!TestMockSupport.RunningMocked)
             {
 #endif
                 if (useAutoSaveProfile)
                 {
-                    ProtectedProfileProvider.InitializeResourceManagerProfile();
+                    try
+                    {
+                        ProtectedProfileProvider.InitializeResourceManagerProfile();
+                    }
+                    catch (SystemException e)
+                    {
+                        if (!(e is IOException) && !(e is UnauthorizedAccessException))
+                        {
+                            throw e;
+                        }
+                        WriteInitializationWarnings(e.Message);
```

Comment:
Please put the warning message to Resources.resx, it should be something like:
"Failed to read azure context profile file with error {0}"